ipa-core/src/test_fixture/ipa.rs

```diff
@@ -253,7 +259,36 @@ pub async fn test_oprf_ipa<F>(
 
     //TODO(richaj): To be removed once the function supports non power of 2 breakdowns
     let _ = result.split_off(expected_results.len());
-    assert_eq!(result, expected_results);
+
+    match dp_params {
+        DpParams::NoDp => {
+            assert_eq!(result, expected_results);
+        }
+        DpParams::WithDp { epsilon } => {
+            let per_user_credit_cap_f64 = f64::from(config.per_user_credit_cap);
+            let noise_params = NoiseParams {
+                epsilon,
+                delta: 1e-6,
+                ell_1_sensitivity: per_user_credit_cap_f64,
+                ell_2_sensitivity: per_user_credit_cap_f64,
+                ell_infty_sensitivity: per_user_credit_cap_f64,
+                ..Default::default()
+            };
+
+            let num_bernoulli = crate::protocol::dp::find_smallest_num_bernoulli(&noise_params);
+            let mean: f64 = f64::from(num_bernoulli) * 0.5; // n * p
```

Comment:
```suggestion
            // Compute the expected value of the noise from the DP parameters.
            let expected_noise: f64 = f64::from(num_bernoulli) * 0.5; // n * p
```

I wonder whether this is something that should be a function on `DpParams`, so that you don't have to use "internal" functions to compute it.  It is, after all, a consequence of how we implement the noise function.

That also suggests that `DpParams::WithDp` would not be the right name, but instead `DpMechanism::Binomial { epsilon: f64 }` instead.

Reply:
Renamed as suggested.  I think putting this function on DpMechanism would still be a bit to high up as there will be differences between `feature_label_dot_product` and histograms.  We could put it as a function on NoiseParams.